linkerd/app/outbound/src/sidecar.rs

```diff
@@ -288,12 +175,120 @@ impl std::hash::Hash for Sidecar {
 
 // === impl HttpSidecar ===
 
+impl From<protocol::Http<Sidecar>> for HttpSidecar {
+    fn from(parent: protocol::Http<Sidecar>) -> Self {
+        let orig_dst = (*parent).orig_dst;
+        let version = svc::Param::<http::Version>::param(&parent);
+        let mut policy = (*parent).policy.clone().expect("policies are required");
+
+        if let Some(mut profile) = (*parent).profile.clone().map(watch::Receiver::from) {
+            // Only use service profiles if there are novel routes/target
+            // overrides.
+            let addr = {
+                let p = profile.borrow();
+                if p.has_routes_or_targets() {
+                    p.addr.clone()
+                } else {
+                    None
+                }
+            };
+            if let Some(addr) = addr {
+                if profile.borrow().has_routes_or_targets() {
+                    tracing::debug!("Using ServiceProfile");
+                    let init = Self::mk_profile_routes(addr.clone(), &*profile.borrow_and_update());
+                    let routes =
+                        http::spawn_routes(profile, init, move |profile: &profiles::Profile| {
+                            Some(Self::mk_profile_routes(addr.clone(), profile))
+                        });
+                    return HttpSidecar {
+                        orig_dst,
+                        version,
+                        routes,
+                    };
+                }
+            }
+        }
```

Comment:
this feels needlessly convoluted --- `addr` is only `Some` if `profile.has_routes_or_targets()`, why do we need the `Option`? can't we just write
```suggestion
            // Only use service profiles if there are novel routes/target
            // overrides.
            if profile.borrow().has_routes_or_targets() {
                let addr = profile.borrow().addr.clone()
                tracing::debug!("Using ServiceProfile");
                let init = Self::mk_profile_routes(addr.clone(), &*profile.borrow_and_update());
                let routes =
                    http::spawn_routes(profile, init, move |profile: &profiles::Profile| {
                        Some(Self::mk_profile_routes(addr.clone(), profile))
                    });
                return HttpSidecar {
                    orig_dst,
                    version,
                    routes,
                };
            }
```

Reply:
oh i see that you just fixed that, whoops.

Reply:
basically, this makes sure that we have a logicaladdr that we can reuse no matter the state of the profile.